shared/js/background/dynamic-rule-id.js

```diff
@@ -0,0 +1,40 @@
+const browserWrapper = require('./wrapper.es6')
+const DYNAMIC_RULE_ID_START = 100000
```

Comment:
I'm wondering if this dynamic rule ID logic should be in the `declarative-net-request.js` module. But if not, I think that:
1. we should export the `DYNAMIC_RULE_ID_START` constant from the `delcarative-net-request.js` module and import it here. We should add a comment in  `declarative-net-request.js` for the constant to mention that rule IDs > 100k are reserved for dynamic rule IDs, and to reference dynamic-rule-is.js.
2. We should add DNR or similar to the name of this module, just so it's a bit clearer what the module does from the name. E.g. how about `dynamic-dnr-rule-id.js`? (While working on this, I often found the term "rule" confusing since it could be a tracker blocking rule, or a declarativeNetRequest rule... so good to specify where possible.)
3. We should add a comment near the top of this module to explain a bit what the point of all this dynamic rule ID stuff is.

Reply:
1. I've had timing issues using `delcarative-net-request.js`. Seems like it's used elsewhere on startup and requiring it in other areas can cause errors.
2. 👍 
3. 👍